Replacer/Program.cs

```
@@ -41,73 +41,183 @@ private static bool RemoteFileExists(string url)
                 //Any exception will returns false.
                 return false;
             }
+        }*/
+
+        private static async Task<string> GetJsonString(string url)
+        {
+            HttpClient client = new HttpClient();
+            HttpResponseMessage response = await client.GetAsync(url);
+            if (response.IsSuccessStatusCode)
+            {
+                //return await Task.FromResult(response.Content.ReadAsStringAsync());
+                client.Dispose();
+                return await response.Content.ReadAsStringAsync();
+
+            }
+            else
+            {
+                throw new Exception($"Запрос не выполнен. Код состояния: {response.StatusCode}");
+            }
+        }
+
+        private static List<Replacement> GetReplacements()//получение замен
+        {
+            if (File.Exists(@"replacement.json"))
```

я думаю, что File.ReadAllText и так выкинет ошибку, если файла не существовало. Это круто, что ты задумываешься об обработке ошибок, но я думаю, что здесь это немного излишне

Можно оставить эту инструкцию, но комбинацию if/else изменить. Есть такое понятие как [fail early и if-guard](https://www.tutisani.com/software-architecture/nested-if-vs-guard-condition.html).
Такой вариант сильно проще читать

```CSharp
if(!File.Exists) throw

// normal flow
string text = File.ReadAllText(@"replacement.json");
return JsonConvert.DeserializeObject<List<Replacement>>(text);
```